force_wfmanager/left_side_pane/data_source_model_view.py

```diff
@@ -260,12 +260,21 @@ def update_slot_info_names(self):
                              self.output_slots_representation):
             row.name = info.name
 
-    @on_trait_change('variable_names_registry.available_variables[]')
+    @on_trait_change('variable_names_registry.available_variables[],'
+                     'output_slots_representation.name,'
+                     'output_slots_representation.type,'
+                     'input_slots_representation.name,'
+                     'input_slots_representation.type,')
```

Comment:
Careful with this end comma. It might not lead to any problems but it's hard to say.